docs/en/operations/system-tables/information_schema.md

```diff
@@ -218,3 +226,62 @@ is_trigger_updatable:       NO
 is_trigger_deletable:       NO
 is_trigger_insertable_into: NO
 ```
+
+## KEY_COLUMN_USAGE (#key_column_usage)
+
+It was added for compatibility with third party tools such as Tableau Online. Contains only the primary keys columns read from [system.columns](../../operations/system-tables/columns.md).
+
+Columns:
```

Comment:
Could you please check the exact field types that MySQL uses for this view, e.g. here: https://onecompiler.com/mysql/3zmzfnxtm

We should generally aim to provide the same types. Basically where we can't provide a reasonable value (e.g. `referenced_column_name`), we should simply return "" instead of NULL (having type `String` instead of `Nullable(String)`).

Reply:
So, we have this definition of the table:

```
+-------------------------------+--------------+------+-----+---------+-------+
| Field                         | Type         | Null | Key | Default | Extra |
+-------------------------------+--------------+------+-----+---------+-------+
| CONSTRAINT_CATALOG            | varchar(64)  | NO   |     | NULL    |       |
| CONSTRAINT_SCHEMA             | varchar(64)  | NO   |     | NULL    |       |
| CONSTRAINT_NAME               | varchar(64)  | YES  |     | NULL    |       |
| TABLE_CATALOG                 | varchar(64)  | NO   |     | NULL    |       |
| TABLE_SCHEMA                  | varchar(64)  | NO   |     | NULL    |       |
| TABLE_NAME                    | varchar(64)  | NO   |     | NULL    |       |
| COLUMN_NAME                   | varchar(64)  | YES  |     | NULL    |       |
| ORDINAL_POSITION              | int unsigned | NO   |     | 0       |       |
| POSITION_IN_UNIQUE_CONSTRAINT | int unsigned | YES  |     | NULL    |       |
| REFERENCED_TABLE_SCHEMA       | varchar(64)  | YES  |     | NULL    |       |
| REFERENCED_TABLE_NAME         | varchar(64)  | YES  |     | NULL    |       |
| REFERENCED_COLUMN_NAME        | varchar(64)  | YES  |     | NULL    |       |
+-------------------------------+--------------+------+-----+---------+-------+
```

which seems to be an exact match to what we have here:

```
ATTACH VIEW key_column_usage
(
     `constraint_catalog` String,
     `constraint_schema` String,
     `constraint_name` Nullable(String),
     `table_catalog` String,
     `table_schema` String,
     `table_name` String,
     `column_name` Nullable(String),
     `ordinal_position` UInt32,
     `position_in_unique_constraint` Nullable(UInt32),
     `referenced_table_schema` Nullable(String),
     `referenced_table_name` Nullable(String),
     `referenced_column_name` Nullable(String),
     ...
```

isn't it?

Reply:
MySQL is OK with returning nulls.

```
show create table t1;
+-------+------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
| Table | Create Table                                                                                                                                                                       |
+-------+------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
| t1    | CREATE TABLE `t1` (
  `s1` int DEFAULT NULL,
  `s2` int DEFAULT NULL,
  `s3` int NOT NULL,
  PRIMARY KEY (`s3`)
) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_0900_ai_ci |
+-------+------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+


select * from information_schema.key_column_usage where table_name = 't1';
+--------------------+-------------------+-----------------+---------------+--------------+------------+-------------+------------------+-------------------------------+-------------------------+-----------------------+------------------------+
| CONSTRAINT_CATALOG | CONSTRAINT_SCHEMA | CONSTRAINT_NAME | TABLE_CATALOG | TABLE_SCHEMA | TABLE_NAME | COLUMN_NAME | ORDINAL_POSITION | POSITION_IN_UNIQUE_CONSTRAINT | REFERENCED_TABLE_SCHEMA | REFERENCED_TABLE_NAME | REFERENCED_COLUMN_NAME |
+--------------------+-------------------+-----------------+---------------+--------------+------------+-------------+------------------+-------------------------------+-------------------------+-----------------------+------------------------+
| def                | default           | PRIMARY         | def           | default      | t1         | s3          |                1 |                          NULL | NULL                    | NULL                  | NULL                   |
+--------------------+-------------------+-----------------+---------------+--------------+------------+-------------+------------------+-------------------------------+-------------------------+-----------------------+------------------------+